cmd/controller/main.go

```diff
@@ -179,8 +167,40 @@ func main() {
 }
 
 func init() {
-	// These are commented because they're also defined by controller-runtime.
-	// flag.StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
-	// flag.StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
 	flag.StringVar(&experimentalControllers, "experimentalControllers", "", "List of experimental controllers to include in the Knative Controller.")
+	flag.BoolVar(&hardcodedLoggingConfig, "hardCodedLoggingConfig", false, "If true, use the hard coded logging config. It is intended to be used only when debugging outside a Kubernetes cluster.")
+}
+
+func getLoggingConfigOrDie() map[string]string {
```

Comment:
This seems like a generally useful thing. Could maybe move to knative/pkg in the future.

Minor nit: Might be easier to read as a constant map and helper function in a separate file.